product-service/src/dataStore/pg-client.ts

```diff
@@ -0,0 +1,87 @@
+'use strict';
+import { Client } from 'pg';
+
+import {Product} from '../types/Product';
+
+
+const { PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD } = process.env;
+
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false // to avoid warring in this example
+    },
+    connectionTimeoutMillis: 5000 // time in millisecond for termination of the database query
+};
+
+const productFields = 'id, title, description, price, count';
+
+export const getAll = async (): Promise<Product[]> => {
+    const client = new Client(dbOptions);
+    await client.connect();
+
+    try {
+        const script = `select ${productFields} from products, stocks where id = product_id`;
+        console.log(`Script for getAll: ${script}`);
+        const { rows: products } = await client.query(script);
+
+        return products;
+    } catch (error) {
+        console.error('Error during database getAll request executing:', error);
+        throw  error;
+    } finally {
+        client.end();
+    }
+}
+
+export const get = async (id): Promise<Product> => {
+    const client = new Client(dbOptions);
+    await client.connect();
+
+    try {
+        const script = `select ${productFields} from products, stocks where id = '${id}' and id = product_id`;
```

Comment:
Risk of SQL injection. Even if you validate `id` with Joi, this function is independent from it, so by its own behavior this query is vulnerable.

Reply:
What's the best security approach? To check all insert params directly before injection into query?